text/0000-project-groups.md

```diff
@@ -0,0 +1,328 @@
+    pinging specific people from team to gauge their interest.
+  - You may not always be able to find someone who is willing to act as
+    liasion. It's up to each team to decide how many new efforts
+    they'll have the bandwidth for, which may at times be none.
+
+2. Obtain the consensus of the team to create group.
+
+- Specify the liaison, and shepherd(s).
+- Write a short motivation, and some notes on possible solutions.
+- How consensus is reached would vary from team to team, some would require an
+  RFC while others could decide in a meeting. (See [Future Work](#future-work))
+
+3. Create infrastructure for group.
+
+- GitHub repository under `rust-lang` for hosting work and discussions, such
+  as for draft RFCs.
+- A Discord channel or a Zulip stream for communication.
+- Project group in [`rust-lang/team`], as well as a team on GitHub, for
+  handling permissions.
+
+4. Create a post on the Inside Rust blog announcing creation of the group. Be
+   sure to include the following information.
+  - An introduction
+  - The charter (either linked or inlined)
+  - A link to your group's GitHub repository
+  - If your group is open for participants, provide information on how they
+    can contribute.
+    - If you're also planning on running regular meetings, include when your
+      group plans to meet along with a link to calendar event for the meeting.
+
+5. The group works towards the goals laid out in their charter.
+
+6. When active work has stopped a group is "archived".
+- Archival can be started by the project group shepherds.
+- Archival is not necessarily a permanent state, it is only a reflection on the current
+  status of the group. A group can be "restored" at a later stage.
+- Reasons to archive (non-exhaustive):
+  - Nobody in the group has time anymore or higher priority things arose.
+  - There's a blocking issue that can't be resolved.
+  - Don't see any additional work to do in this area in the near future.
+  - The work was done to a satisfactory state.
+  - The group decided the idea wasn't so good after all.
+
+7. Create a blog post announcing the archival of the group.
+
+- The scope of this post will vary based on the scope of the group, but
+  ideally it would include some of the following.
+  - Overview of decisions, RFCs, and other output the group produced.
+  - Thoughts on the process, how it worked (or didn't as case may be), any
+    difficulties encountered, and what they would want to be improved.
+
+8. Archive infrastructure.
+
+- Archive GitHub repository to be read-only.
+- Archive chat channel(s) on any platforms.
+
+9. (Optional) Restore group
+
+- At any later point the group could be restored to active status if there's
+  interest among the team to revive the effort.
+- If significant time has passed, part of restoring the group should be to
+  evaluate whether the past decisions and rationale are still applicable to the
+  present.
+- Go to step 2.
```

Comment:
I continue to be opposed to restoration as an explicit step, I would want a fresh start I think in all cases after archival. I don't really see benefits to restoration vs. restarting; could you elaborate why you find restoring worthwhile?

At the very least we should go to step 1, there should not be an implicit assumption that the same things are motivational today, nor that the liaison previously available still is.

Reply:
This feels like something best left up to the individual teams. I don't think we'll be necessarily following everything to the letter all the time anyways, and may find interpretations that are the best fit for the team and the situation.

Reply:
It seems clear to me that in all cases where a team actively wants to restart a working group, having a restoration process documented isn't necessary: the steps required to start a working group are simple to do if there's interest from the team and available personnel.

If the team does not currently have bandwidth/desire to restart the group, then it seems like we would not want a dedicated process for it.

Mainly, I do not see the advantage of having a dedicated process for restoration; it seems to not benefit us, and I can see it being harmful.

Reply:
Good points. It's stated here that "go to step 2", which doesn't save the team a notable amount of time, and e.g., the language team could just shorten the initial discussion period etc.

Reply:
Yeah, I think that we might be interpreting "Restoration" differently. I wasn't intending for there to be a way to skip a step, saying Restoration is more a way to express that efforts are continual on a larger scale. For example there is a traits working group, but when that group finishes it doesn't mean that traits are "complete" or there won't be future efforts in traits or trait working groups.

Reply:
I think so, yes; I agree with your interpretation but I think the RFC makes such an interpretation not at all clear :)

I would propose that we remove all current mentions of restoration, and instead include something like:

The archival of a project group does not mean that the problem space for that area of Rust has been exhausted.

Does that sound reasonable to you?